gqlalchemy/loaders.py

```diff
@@ -126,16 +128,132 @@ class TableMapping:
     indices: Optional[List[str]] = None
 
 
-class DataSource(ABC):
+@dataclass(frozen=True)
+class NameMappings:
+    """
+    Class that contains new label name and all of the column name mappings for a single table
+
+    :param label: New label (Optional)
+    :type label: str
+    :param column_names_mapping: Dictionary containing key-value pairs in form ("column name", "property name") (Optional)
+    :type column_names_mapping: Dict[str, str]
     """
-    Base class. Inherit this class and encapsulate reading of the file format from some data source
+    label: Optional[str] = None
+    column_names_mapping: Optional[Dict[str, str]] = None
+
+
+class FileSystemHandler(ABC):
     """
-    def __init__(self, file_extension: str) -> None:
+    Abstract class for defining FileSystemHandler. Inherit this class and initialize
+    connection to custom data source.
+    """
+    def __init__(self) -> None:
+        super().__init__()
+
+    @abstractmethod
+    def get_path(self):
+        pass
```

Comment:
Do we need documentation here? Is it unique what get_path represents for all FS?

Reply:
added documentation. No, get_path is there so we can call it with the name of the file we want to read, and it returns the full path in that specific file system

Reply:
:+1: